### DISCUSSION TITLE
Why is babel-plugin-react-compiler required for react 19 compile support - doesn't next use SWC?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#66544] [Help]
### Summary

Why is `babel-plugin-react-compiler` - I thought it would need a SWC plugin.

### Additional information

Why do I ask - is there some way to use babel plugins without going all the way back to babel in full? OR, does enabling experimental: reactCompiler, also put us in babel mode? It's not super clear what's happening here.

There is a swc plugin for graphql called `@graphql-codegen/client-preset-swc-plugin` which [I've never been able to get to work in nextjs](https://github.com/dotansimha/graphql-code-generator/issues/9450), because using SWC plugins is a total nightmare, as is getting babel to work. But if there is some way I can get the babel version of that plugin to work, that'd be swell.

**Comment by damassi** [comment]
Would also love an OSS React 19 compiler SWC plugin!

**Comment by icyJoseph** [comment]
Hi,

The problem in general was that the compiler has been first implemented in JS, for Babel! So in a way, it is an upstream _issue_. What has to happen is that the compiler model, or architecture, stabilizes, and then it can be written in other languages and such.

Source of my claim: https://github.com/facebook/react/blob/main/compiler/docs/DEVELOPMENT_GUIDE.md#wip-rust-development and other documents on that folder of the react repo
